src/actions/index.js

```diff
@@ -0,0 +1,150 @@
+            history.push('/dashboard');
+        }catch(error){
+            dispatch({
+                type: AUTHENTICATED_ERROR,
+                payload: 'Invalid email or password'
+            });
+        }
+    }
+}
+
+// action creator for signing up user
+export const signUpAction = (values, callback) => {
+    return async (dispatch) => {
+        try {
+            const request = await axios.post(`${ROOT_URL}/auth/register`, values);
+            dispatch({
+                type: REGISTER,
+                payload: request
+            });
+            callback();
+        }catch(error) {
+            dispatch({
+                type: USER_ALREADY_EXISTS,
+                payload: 'User already exists'
+            })
+        }
+    }
+}
+
+// logging/signing out user
+export const signOutAction = () => {
+    return async (dispatch) => {
+        try {
+            const request = await instance.post(`${ROOT_URL}/auth/logout`);
+            localStorage.clear();
+            dispatch({
+                type: UNAUTHENTICATED,
+                payload: request
+            });
+        }catch(error){
+            dispatch({
+                type: AUTHENTICATED_ERROR,
+                payload: 'Invalid token, please login again!'
+            });
+        }
+    }
+}
+    
+
+/* action creator for creating new categories 
+    takes in form data values and callback method
+    executed after submition 
+*/
+export const createCategory = (values, callback) => {
+    const request = instance.post(`${ROOT_URL}/recipe_category`, values)
+        .then(() => callback());
+    
+    return {
+        type: CREATE_CATEGORY,
+        payload: request
+    }
+}
+
+export const fetching = () => {
+    return {
+        type: "FETCHING"
+    }
+}
+
+export const fetchCat = (res) => {
+    return {
+        type: FETCH_CATEGORIES,
+        payload: res,
+    }
+}
+
+// action creator for fetching categories from the database
+export const fetchCategories = () => {
+    return async (dispatch) => {
+        try {
+            dispatch(fetching);
+            const request = await instance.get(`${ROOT_URL}/recipe_category`);
+
+            dispatch(fetchCat(request));
+        }catch(error) {
+            dispatch({
+                type: "UNAUTHENTICATED",
+                payload: "Invalid authentication credentials"
+            });
+        }
+    }
+}
+
+export const fetchRec = (res) => {
+    return {
+        type: FETCH_RECIPES,
+        payload: res,
+    }
+}
+// action creator for fetching recipes from the database
+export const fetchRecipes = () => {
+    return async (dispatch) => {
+        try {
+            dispatch(fetching);
+            const request = await instance.get(`${ROOT_URL}/recipes`);
+
+            dispatch(fetchRec(request));
+        }catch(error) {
+            dispatch({
+                type: "UNAUTHENTICATED",
+                payload: "Invalid authentication credentials"
+            });
+        }
+    }
+}
```

Comment:
I think it would be better if you seperated your actions into seperate files e.g. auth.js, category.js etc. Your code will be a lot more readable and things will also be easier to find.